{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/training/notebooks/Train.py

```diff
@@ -1,140 +1,137 @@
-#
-#
-# Return values:
-# * model_uri: The notebook must return the URI of the registered model as notebook output specified through
-#              dbutils.notebook.exit() AND as a task value with key "model_uri" specified through
-#              dbutils.jobs.taskValues(...), for use by downstream notebooks.
-#
-# For details on MLflow Recipes and the individual split, transform, train etc steps below, including usage examples,
-# see the Regression Recipe overview documentation: https://mlflow.org/docs/latest/recipes.html
-# and Regression Recipes API documentation: https://mlflow.org/docs/latest/python_api/mlflow.recipes.html
+# * env (required):                 - Environment the notebook is run in (staging, or prod). Defaults to "staging".
+# * training_data_path (required)   - Path to the training data.
+# * experiment_name (required)      - MLflow experiment name for the training runs. Will be created if it doesn't exist.
+# * model_name (required)           - MLflow registered model name to use for the trained model. Will be created if it
+# *                                   doesn't exist.
 ##################################################################################
 
-# COMMAND ----------
-# MAGIC %load_ext autoreload
-# MAGIC %autoreload 2
-
-# COMMAND ----------
-
-import os
-notebook_path =  '/Workspace/' + os.path.dirname(dbutils.notebook.entry_point.getDbutils().notebook().getContext().notebookPath().get())
-%cd $notebook_path
-
-# COMMAND ----------
-
-# MAGIC %pip install -r ../../requirements.txt
-
-# COMMAND ----------
-
-dbutils.library.restartPython()
-
-# COMMAND ----------
-
-import os
-notebook_path =  '/Workspace/' + os.path.dirname(dbutils.notebook.entry_point.getDbutils().notebook().getContext().notebookPath().get())
-%cd $notebook_path
-%cd ../
 
 # COMMAND ----------
+# DBTITLE 1, Notebook arguments
 
-from mlflow.recipes import Recipe
+# List of input args needed to run this notebook as a job.
+# Provide them via DB widgets or notebook arguments.
 
-try:
-    # Profile from databricks workflow input will be used.
-    env = dbutils.widgets.get("env")
-    profile = f"databricks-{env}"
-except:
-    # When the users manually run the notebook, we use the "databricks-dev" profile instead of staging, prod or test.
-    profile = "databricks-dev"
+# Notebook Environment
+dbutils.widgets.dropdown("env", "staging", ["staging", "prod"], "Environment Name")
+env = dbutils.widgets.get("env")
 
-# COMMAND ----------
+# Path to the Hive-registered Delta table containing the training data.
+dbutils.widgets.text(
+    "training_data_path",
+    "/databricks-datasets/nyctaxi-with-zipcodes/subsampled",
+    label="Path to the training data",
+)
 
-from mlflow.recipes.utils import (
-    get_recipe_config,
-    get_recipe_root_path,
+# MLflow experiment name.
+dbutils.widgets.text(
+    "experiment_name",
+    f"/dev-{{cookiecutter.experiment_base_name}}",
```

Comment:
Do we really want to hardcode the `dev-` prefix here?

Reply:
This is the default value when users manually run the notebook. We want to make it clear that it's not for CI/CD/staging/prod purpose. That's why we added the `dev-`
If it's from CI/CD, the experiment name parameter is passed to the notebook and will overwrite this default value.